### ISSUE TITLE
[libc++] filesystem::remove_all returns wrong error code on a read-only nested directory

### ISSUE THREAD

**Issue by jakeegan** [llvm/llvm-project#197100]
Originally reported here: https://github.com/nodejs/node/issues/62790

Test case:
```C++
#include <iostream>
#include <filesystem>
#include <system_error>
#include <sys/stat.h>

int main() {
    std::filesystem::create_directories("parent/child");
    
    // Make parent read-only to trigger permission error
    chmod("parent", 0555);
    
    std::error_code error;
    std::filesystem::remove_all("parent", error);
    
    std::cout << "error.value() = " << error.value() << std::endl;
    std::cout << "error.message() = " << error.message() << std::endl;
    
    chmod("parent", 0755);
    std::filesystem::remove_all("parent");

    return 0;
}
```
AIX Output:
```
error.value() = 17
error.message() = File exists
```
MacOS Output:
```
error.value() = 66
error.message() = Directory not empty
```

It's expected for `remove_all` to fail with a permission error rather than the above errors.
